sections/header.liquid

```diff
@@ -437,15 +437,21 @@
       this.header = document.getElementById('shopify-section-header');
       this.headerBounds = {};
       this.currentScrollTop = 0;
+      this.preventReveal = false;
+
+      window.preventHeaderReveal = new Event('preventHeaderReveal', {'bubbles': true});
 
       this.onScrollHandler = this.onScroll.bind(this);
+      this.hideHeaderOnScrollUp = this.setPreventReveal.bind(this);
 
+      document.addEventListener('preventHeaderReveal', this.hideHeaderOnScrollUp);
```

Comment:
I'm not so sure about this `preventHeaderReveal` listener on a `document` level + use of bubbling, because that means the event is being propagated on multiple levels...

Sure, from my limited knowledge, it seems like the performance impact is probably insignificant. But from a code design perspective, I feel like this event should be dispatched directly to the `sticky-header` from `VariantSelects` so this `StickyHeader` JS can easily add the eventListeners to itself.

Thoughts?